### PR TITLE
fix: display workspace avatars correctly when URLs fail to load

### PR DESCRIPTION
**Pull request by buenos-nachos**
No issue to link – this was an issue that surfaced for certain URL types when kicking the tires on the orgs work

## Changes made
- Updated custom avatar components to favor background color by default
- Updated `AvatarData` component to let you manually specify the source of the text used when images fail to load, and updated the orgs breadcrumb segment to use it
- Added some logic for handling emoji images better

### CODE REVIEW

**Comment by matifali** [comment]
@Parkreiner, there are a few changes in UI tests.

**Comment by Parkreiner** [comment]
Making a quick comment for documentation purposes: Chromatic is currently down, which is blocking part of the CI pipeline. I could try to push the changes through as-is, but since the previous version of this PR broke parts of the UI, I want to err on the side of caution

Will try to re-run processes right before I log off today, but this might have to wait until Monday

File: site/src/components/AvatarData/AvatarData.tsx [hunk 1]
```diff
@@ -8,18 +8,35 @@ export interface AvatarDataProps {
...
+	imgFallbackText,
 	avatar,
 }) => {
 	const theme = useTheme();
-
 	if (!avatar) {
-		avatar = <Avatar src={src}>{title}</Avatar>;
+		avatar = (
+			<Avatar background src={src}>
+				{typeof title === "string" ? title || "-" : (imgFallbackText ?? "-")}
```
Comment (Parkreiner) [inline site/src/components/AvatarData/AvatarData.tsx:37]: Wish Biome's formatter were more consistent, and added some parentheses for the `title || "-"` part

File: site/src/pages/ManagementSettingsPage/SidebarView.tsx
Comment (Parkreiner) [inline site/src/pages/ManagementSettingsPage/SidebarView.tsx:1]: Nothing about the logic here changed – just got formatted differently

File: site/src/components/AvatarData/AvatarData.tsx [hunk 2]
```diff
@@ -8,18 +8,35 @@ export interface AvatarDataProps {
...
+	imgFallbackText,
 	avatar,
 }) => {
 	const theme = useTheme();
-
 	if (!avatar) {
-		avatar = <Avatar src={src}>{title}</Avatar>;
+		avatar = (
+			<Avatar background src={src}>
+				{typeof title === "string" ? title || "-" : imgFallbackText ?? "-"}
```
Comment (aslilac) [inline site/src/components/AvatarData/AvatarData.tsx:37]:
```suggestion
				{(typeof title === "string" ? title : imgFallbackText) || "-"}
```

File: site/src/components/AvatarData/AvatarData.tsx [hunk 3]
```diff
@@ -8,18 +8,35 @@ export interface AvatarDataProps {
...
+	 * Useful for when you need to pass in a ReactNode for the title of the
+	 * component.
+	 *
+	 * MUI will try to take any string titles and turn them into the first
+	 * character, but if you pass in a ReactNode, MUI can't do that, because it
+	 * has no way to reliably grab the text content during renders.
+	 *
+	 * Tried writing some layout effect/JSX parsing logic to do the extraction,
+	 * but it added complexity and render overhead, and wasn't reliable enough
+	 */
```
Comment (aslilac) [inline site/src/components/AvatarData/AvatarData.tsx:22]:
```suggestion
	/**
	 * Useful for when you need to pass in a `ReactNode` for the title of the
	 * component, but you still want to use the first character of a bit of text
	 * as the fallback.
	 */
```

Comment (aslilac) [inline site/src/components/AvatarData/AvatarData.tsx:22]:
```suggestion
	/**
	 * Useful for when you need to pass in a `ReactNode` for the title of the
	 * component, but you still want to use the first character of a bit of text
	 * as the fallback.
	 */
```

File: site/src/pages/ManagementSettingsPage/SidebarView.tsx
Comment (aslilac) [reply site/src/pages/ManagementSettingsPage/SidebarView.tsx:1]: thank you 😄

File: site/src/utils/appearance.ts [hunk 4]
```diff
@@ -14,3 +14,18 @@ export const getLogoURL = (): string => {
...
+ * Helps when you need to style emojis differently (i.e., not adding rounding to
+ * the container so that the emoji doesn't get cut off).
+ */
+export function isEmojiUrl(url: string | undefined): boolean {
+	if (!url) {
+		return false;
+	}
+
+	return url.startsWith("/emojis/");
+}
```
Comment (aslilac) [inline site/src/utils/appearance.ts:31]: hmm. I'm skeptic about adding special handling like this, but I get the motivation.

**Review by aslilac — APPROVED** [review]

File: site/src/components/AvatarData/AvatarData.tsx (hunk 2 above)
Comment (Parkreiner) [reply site/src/components/AvatarData/AvatarData.tsx:37]: Oh, good call

**Pull request merged**